src/components/NewDesign/Modals/SelectModals/SelectTokenModal/SelectTokenModal.tsx

```diff
@@ -84,33 +95,34 @@ export const SelectTokenModal: React.FC<ISelectTokenModal> = ({
         </Grid>
         <Box className={classes.tokenList}>
           <CustomScrollbar>
-            {tokens.filter(token => {
-              return token.symbol.toLowerCase().includes(value)
-            }).map((token, index) => (
+            { tokens ? tokens.filter(token => {
+              return token ? token.symbol.toLowerCase().includes(value)
+                : null
+            }).map((token) => (
               <Grid
                 container
-                key={`tokens-${token.symbol}`}
+                key={token ? `tokens-${token.symbol}` : ''}
                 className={classes.tokenItem}
                 alignItems='center'
                 wrap='nowrap'
                 onClick={() => {
-                  onSelect(index)
+                  onSelect(tokenIndex(token ? token.symbol : ''))
                   handleClose()
                 }}>
                 <Grid item>
                   <CardMedia
                     className={classes.tokenIcon}
-                    image={icons[token.symbol] ?? icons.USDT}
+                    image={token ? icons[token.symbol] : ''}
                   />{' '}
                 </Grid>
-                <Grid item className={classes.tokenData}>
-                  <Typography className={classes.tokenName}>{token.symbol}</Typography>
+                <Grid item>
+                  <Typography className={classes.tokenName}>{token ? token.symbol : ''}</Typography>
                   <Typography className={classes.tokenDescrpiption}>
-                    {descrpitionForSymbol[token.symbol] ?? 'Asset'}
+                    {token ? descrpitionForSymbol[token.symbol] ?? 'Asset' : ''}
```

Comment:
use here name from tokens list or some prop

Reply:
first of all i need to connect solana token list to Redux

Reply:
just mock this in story